code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -207,130 +169,97 @@ This saves us from having to call add_fingerprint() any time something is put in
+			W.screen_loc = find_inv_position(slot)
+			W.layer = ABOVE_HUD_LAYER
+			W.plane = ABOVE_HUD_PLANE
+
+			// That's really reqed. At least for now
+			if(client)
+				client.screen |= W
+
+			if(W.action_button_name)
+				update_action_buttons()
 
-//This is an UNSAFE proc. Use mob_can_equip() before calling this one! Or rather use equip_to_slot_if_possible() or advanced_equip_to_slot_if_possible()
+
+//This is an UNSAFE proc. Use mob_can_equip() before calling this one! Or rather use equip_to_slot_if_possible()
 //set redraw_mob to 0 if you don't wish the hud to be updated - if you're doing it manually in your own proc.
-/mob/living/carbon/human/equip_to_slot(obj/item/W as obj, slot, redraw_mob = 1)
-
-	if(!slot) return
-	if(!istype(W)) return
-	if(!species || !species.hud || !(slot in species.hud.equip_slots)) return
-	if(ismob(W.loc))
-		var/mob/M = W.loc
-		if(M.get_inventory_slot(W) && !M.prepare_for_slotmove(W))
-			return
-	W.forceMove(src)
+/mob/living/carbon/human/proc/legacy_equip_to_slot(obj/item/W, slot, redraw_mob = 1)
 	switch(slot)
 		if(slot_back)
 			src.back = W
-			W.equipped(src, slot)
-			update_inv_back(redraw_mob)
 		if(slot_wear_mask)
 			src.wear_mask = W
 			if(wear_mask.flags_inv & (BLOCKHAIR|BLOCKHEADHAIR))
 				update_hair(redraw_mob)	//rebuild hair
 				update_inv_ears(0)
-			W.equipped(src, slot)
-			update_inv_wear_mask(redraw_mob)
 		if(slot_handcuffed)
 			src.handcuffed = W
-			update_inv_handcuffed(redraw_mob)
 		if(slot_legcuffed)
 			src.legcuffed = W
-			W.equipped(src, slot)
-			update_inv_legcuffed(redraw_mob)
 		if(slot_l_hand)
 			src.l_hand = W
-			W.equipped(src, slot)
-			update_inv_l_hand(redraw_mob)
 		if(slot_r_hand)
 			src.r_hand = W
-			W.equipped(src, slot)
-			update_inv_r_hand(redraw_mob)
 		if(slot_belt)
 			src.belt = W
-			W.equipped(src, slot)
-			update_inv_belt(redraw_mob)
 		if(slot_wear_id)
 			src.wear_id = W
-			W.equipped(src, slot)
-			update_inv_wear_id(redraw_mob)
 		if(slot_l_ear)
 			src.l_ear = W
 			if(l_ear.slot_flags & SLOT_TWOEARS)
 				var/obj/item/clothing/ears/offear/O = new(W)
 				equip_to_slot_if_possible(O, slot_r_ear, TRUE, FALSE, FALSE)
-			W.equipped(src, slot)
-			update_inv_ears(redraw_mob)
 
 		if(slot_r_ear)
 			src.r_ear = W
 			if(r_ear.slot_flags & SLOT_TWOEARS)
 				var/obj/item/clothing/ears/offear/O = new(W)
 				equip_to_slot_if_possible(O, slot_l_ear, TRUE, FALSE, FALSE)
-			W.equipped(src, slot)
-			update_inv_ears(redraw_mob)
 		if(slot_glasses)
 			src.glasses = W
-			W.equipped(src, slot)
-			update_inv_glasses(redraw_mob)
 		if(slot_gloves)
 			src.gloves = W
-			W.equipped(src, slot)
-			update_inv_gloves(redraw_mob)
 		if(slot_head)
 			src.head = W
 			if(head.flags_inv & (BLOCKHAIR|BLOCKHEADHAIR|HIDEMASK))
 				update_hair(redraw_mob)	//rebuild hair
 				update_inv_ears(0)
 				update_inv_wear_mask(0)
-			W.equipped(src, slot)
-			update_inv_head(redraw_mob)
 		if(slot_shoes)
 			src.shoes = W
-			W.equipped(src, slot)
```

Comment:
You've removed a lot of calls to equipped proc here. Have they been moved elsewhere?